tests/documents/rules/test_rules_comment_chapter.py

```diff
@@ -17,96 +19,157 @@ def test_perm_exists():
         module__project__access=Access.SEMIPUBLIC)
+
     anonymous, moderator, initiator = setup_users(project)
-    participant = user2
+    participant = user_factory()
     project.participants.add(participant)
+    group_member_in_orga, group_member_out, group_member_in_project, project \
+        = setup_group_users(user_factory, group_factory, project)
 
     assert project.access == Access.SEMIPUBLIC
     with freeze_phase(phase):
         assert not rules.has_perm(perm_name, anonymous, item)
         assert not rules.has_perm(perm_name, user, item)
+        assert not rules.has_perm(perm_name, group_member_in_orga, item)
+        assert not rules.has_perm(perm_name, group_member_out, item)
+        assert rules.has_perm(perm_name, group_member_in_project, item)
         assert rules.has_perm(perm_name, participant, item)
         assert rules.has_perm(perm_name, moderator, item)
         assert rules.has_perm(perm_name, initiator, item)
 
 
 @pytest.mark.django_db
-def test_phase_active_project_draft(phase_factory, chapter_factory, user):
+def test_phase_active_project_draft(phase_factory, chapter_factory,
+                                    user_factory, group_factory,
+                                    user):
     phase, _, project, item = setup_phase(phase_factory, chapter_factory,
                                           phases.CommentPhase,
                                           module__project__is_draft=True)
+
     anonymous, moderator, initiator = setup_users(project)
+    group_member_in_orga, group_member_out, group_member_in_project, project \
+        = setup_group_users(user_factory, group_factory, project)
 
     assert project.is_draft
     with freeze_phase(phase):
         assert not rules.has_perm(perm_name, anonymous, item)
         assert not rules.has_perm(perm_name, user, item)
+        assert not rules.has_perm(perm_name, group_member_in_orga, item)
+        assert not rules.has_perm(perm_name, group_member_out, item)
+        assert rules.has_perm(perm_name, group_member_in_project, item)
         assert rules.has_perm(perm_name, moderator, item)
         assert rules.has_perm(perm_name, initiator, item)
 
 
 @pytest.mark.django_db
-def test_post_phase_project_archived(phase_factory, chapter_factory, user):
+def test_post_phase_project_archived(phase_factory, chapter_factory,
+                                     user_factory, group_factory,
+                                     organisation):
+    anonymous = AnonymousUser()
+    user = user_factory()
+    initiator = user_factory()
+    moderator = user_factory()
+
+    group1 = group_factory()
+    group2 = group_factory()
+    group3 = group_factory()
+    group4 = group_factory()
+    group_member_in_orga = user_factory.create(groups=(group1, group2))
+    group_member_out = user_factory.create(groups=(group2, group3))
+    group_member_in_project = user_factory.create(groups=(group2, group4))
+
```

Comment:
Didn't the group and user setup work here or did you overlook this?

Reply:
Oh no, missed that one! Thanks, I'll change it.